pipelines/templates/tasks/upmpackages.yml

```diff
@@ -1,22 +1,41 @@
 # [Template] Builds and publishes UPM .tgz packages
 parameters:
+  projectRoot: $(Get-Location)
+  outputDirectory: $(Build.ArtifactStagingDirectory)\build\upm\output
   version: $(MRTKVersion)
   buildNumber: $(Build.BuildNumber)
+  excludeBuildNumber: 0
 
 steps:
 - task: NodeTool@0
   inputs:
     versionSpec: '12.18.0'
-- task: PowerShell@2
-  displayName: 'Build UPM packages'
-  inputs:
-    targetType: filePath
-    filePath: ./scripts/packaging/createupmpackages.ps1
-    arguments: >
-      -ProjectRoot $(Get-Location)
-      -OutputDirectory $(Build.ArtifactStagingDirectory)\build\upm\output
-      -Version ${{ parameters.version }}
-      -BuildNumber ${{ parameters.buildNumber }}
+    
+- ${{ if eq(parameters.excludeBuildNumber, 0) }}:
+  - task: PowerShell@2
+    displayName: 'Build PREVIEW UPM packages'
+    inputs:
+      targetType: filePath
+      filePath: ./scripts/packaging/createupmpackages.ps1
+      arguments: >
+        -ProjectRoot ${{ parameters.projectRoot }}
+        -OutputDirectory ${{ parameters.outputDirectory }}
+        -Version ${{ parameters.version }}
+        -BuildNumber ${{ parameters.buildNumber }}
+
+- ${{ if eq(parameters.excludeBuildNumber, 1) }}:
+  - task: PowerShell@2
+    displayName: 'Build OFFICIAL UPM packages'
+    inputs:
+      targetType: filePath
+      filePath: ./scripts/packaging/createupmpackages.ps1
+      arguments: >
+        -ProjectRoot ${{ parameters.projectRoot }}
+        -OutputDirectory ${{ parameters.outputDirectory }}
+        -Version ${{ parameters.version }}
+        -BuildNumber ${{ parameters.buildNumber }}
+        -ExcludeBuildNumber
```

Comment:
Since these two tasks are almost identical, is there a way to just use the `${{ if` clause to include/exclude `-ExcludeBuildNumber`? I think the display name is the only other difference, as far as I can tell

Reply:
i've tried once before and dont recall it working. i will try again today

Reply:
cannot add a conditional within a string (arguments is a string). same appears true for the parameters block.

Reply:
I've had similar issues with other parts of the pipeline as well - generally it's OK as long as it ends up being a single thing that is different, but obviously once you have two or more it becomes 2^n configurations which is nasty.

It feels like there should be some solution here but I'm also okay deferring that part.